tests/utils/src/lib.rs

```diff
@@ -0,0 +1,117 @@
+// This file is part of Astar.
+
+// Copyright (C) 2019-2023 Stake Technologies Pte.Ltd.
+// SPDX-License-Identifier: GPL-3.0-or-later
+
+// Astar is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Astar is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Astar. If not, see <http://www.gnu.org/licenses/>.
+
+use frame_support::{traits::Currency, weights::Weight};
+use pallet_contracts_primitives::{Code, ReturnFlags};
+use parity_scale_codec::Decode;
+use sp_runtime::traits::Hash;
+
+type ContractBalanceOf<T> = <<T as pallet_contracts::Config>::Currency as Currency<
+    <T as frame_system::Config>::AccountId,
+>>::Balance;
+
+/// Load a given wasm module from wasm binary contents along
+/// with it's hash.
+///
+/// The fixture files are located under the `../ink-contracts/` directory.
+pub fn load_wasm_module<T>(
+    fixture_name: &str,
+) -> std::io::Result<(Vec<u8>, <T::Hashing as Hash>::Output)>
+where
+    T: frame_system::Config,
+{
+    let fixture_path = ["../ink-contracts/", fixture_name, ".wasm"].concat();
+    let wasm_binary = std::fs::read(fixture_path)?;
+    let code_hash = T::Hashing::hash(&wasm_binary);
+    Ok((wasm_binary, code_hash))
+}
+
+// Load and deploy the contract from wasm binary
+/// and check for successful deploy
+pub fn deploy_wasm_contract<T: pallet_contracts::Config>(
+    contract_name: &str,
+    origin: T::AccountId,
+    value: ContractBalanceOf<T>,
+    gas_limit: Weight,
+    storage_deposit_limit: Option<ContractBalanceOf<T>>,
+    data: Vec<u8>,
+) -> (T::AccountId, <T::Hashing as Hash>::Output) {
```

Comment:
It seems the second value returned is not used in tests?

Reply:
Yeah, they are not used in current tests but are very helpful for debugging while writing tests.